src/Application/UI/Presenter.php

```diff
@@ -1055,19 +1058,21 @@ public static function argsToParams($class, $method, & $args, $supplemental = ar
 
 	/**
 	 * Invalid link handler. Descendant can override this method to change default behaviour.
+	 * @param InvalidLinkException $e
 	 * @return string
 	 * @throws InvalidLinkException
 	 */
 	protected function handleInvalidLink(InvalidLinkException $e)
 	{
-		if ($this->invalidLinkMode === self::INVALID_LINK_SILENT) {
-			return '#';
-
-		} elseif ($this->invalidLinkMode === self::INVALID_LINK_WARNING) {
-			return '#error: ' . $e->getMessage();
-
-		} else { // self::INVALID_LINK_EXCEPTION
-			throw $e;
+		switch($this->invalidLinkMode) {
+			case self::INVALID_LINK_WARNING:
+				trigger_error('Invalid link: ' . $e->getMessage(), E_USER_WARNING);
+			case self::INVALID_LINK_VISUAL:
+				return '#error: ' . $e->getMessage();
+			case self::INVALID_LINK_SILENT:
+			default:
+				trigger_error('Invalid link: ' . $e->getMessage(), E_USER_WARNING);
```

Comment:
This is described in the RFC. SILENT is meant for production and the name refers to the text it returns. E_USER_WARNING is produced so it's visible in the logs.